addons/website_event/controllers/main.py

```diff
@@ -286,7 +286,7 @@ def registration_confirm(self, event, **post):
 
     @http.route(['''/event/<model("event.event", "[('website_id', 'in', (False, current_website_id))]"):event>/ics'''], type='http', auth="public", website=True)
     def make_event_ics_file(self, event, **kwargs):
-        if not event or not event.registration_ids:
+        if not event or not event.sudo().registration_ids:
```

Comment:
Adding `sudo` in public route: :rotating_light:

Reply:
- ``if not event`` part is pointless, if ``event`` cannot be resolved, ``@http.route`` would return a 404 anyway
- If don't understand ``not event.registration_ids`` from a functional point of view... there is no problem in generating an ``.ics`` file for a public event that has no registrations (yet)... by the way, the button we use to generate the ``.ics`` file is displayed after the registration ;-)
=> I would simply remove both checks.
cc @nim-odoo

Reply:
I'm still wondering if `_get_ics_file` doesn't generate any `AccessError` as well in your use case. If `event.registration_ids` causes an error, I don't see why `event.date_begin` doesn't trigger one as well.

Reply:
@nim-odoo 

I think it is because we are trying to read a One2Many field, so we are blocked with ACL on event.registration here, and not directly on event.event